core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -307,20 +307,22 @@ public Condition newCondition() {
         private final Operation.TYPE operationType;
         private final long version;
         private final long seqNo;
+        private final long primaryTerm;
         private final Exception failure;
         private final SetOnce<Boolean> freeze = new SetOnce<>();
         private Translog.Location translogLocation;
         private long took;
 
-        protected Result(Operation.TYPE operationType, Exception failure, long version, long seqNo) {
+        protected Result(Operation.TYPE operationType, Exception failure, long version, long seqNo, long primaryTerm) {
             this.operationType = operationType;
             this.failure = failure;
             this.version = version;
             this.seqNo = seqNo;
+            this.primaryTerm = primaryTerm;
         }
 
-        protected Result(Operation.TYPE operationType, long version, long seqNo) {
-            this(operationType, null, version, seqNo);
+        protected Result(Operation.TYPE operationType, long version, long seqNo, long primaryTerm) {
+            this(operationType, null, version, seqNo, primaryTerm);
```

Comment:
I'm on sure about this one - the semantics we currently have is that Result represents everything that the engine decides. The primary term is part of the operation - I'm not sure we need this here?